distributed/cfexecutor.py

```diff
@@ -127,6 +127,10 @@ def map(self, fn, *iterables, **kwargs):
             raise TypeError("unexpected arguments to map(): %s" % sorted(kwargs))
 
         fs = self._client.map(fn, *iterables, **self._kwargs)
+        if isinstance(fs, list):
+            # Below iterator relies on this being a generator to cancel
+            # remaining futures
+            fs = (val for val in fs)
```

Comment:
https://github.com/dask/distributed/pull/5177

Reply:
This chunk should disappear after merging from main